taichi/runtime/llvm/runtime_module/runtime.cpp

```diff
@@ -561,10 +562,13 @@ struct LLVMRuntime {
   Ptr ambient_elements[taichi_max_num_snodes];
   Ptr temporaries;
   RandState *rand_states;
-  Ptr allocate_aligned(std::size_t size,
+  Ptr allocate_aligned(PreallocatedMemoryChunk &memory_chunk,
```

Comment:
super nits: can we add docstrings for `allocate_aligned` and `allocate_from_reserved_memory`? ;)